### DISCUSSION TITLE
Detecting Platform used to generate/store the passkey (iCloud, Google Key, Windows Hello)

### DISCUSSION THREAD

**Discussion by ChxGuillaume** [MasterKale/SimpleWebAuthn#387] [Troubleshooting]
Hi everyone,

I gave a look at the documentation and different posts on GitHub and couldn't find any information regarding the possibility to detect which "platform" was used to generate the passkey.

I saw the [fmt "AttestationFormat"](https://github.com/MasterKale/SimpleWebAuthn/blob/7cbc1dc6723294874fe4f576c2d32d07337ec5af/packages/server/src/helpers/decodeAttestationObject.ts#L12) option but after testing it most of the time returns me "none" or "package" only 😢 

I know there is the ["credentialDeviceType"](https://api-docs.simplewebauthn.dev/types/_simplewebauthn_typescript_types.credentialdevicetype) that allows you to know if it is saved on a keychain or not (iCloud / Google) but still isn't descriptive enough.

![alt text](https://imgur.com/c8T5EPd.png)

Any idea if that is possible using SimpleWebAuthn's library?

**Comment by MasterKale** [comment]
Hey @ChxGuillaume I'm at Identiverse this week so I'll be a bit slow to respond. I definitely have _thoughts_ about this, though, so I'll follow up as soon as I can :v:
